### PR TITLE
Update Prometheus helm chart to 15.3.0

### PR DESCRIPTION
**Pull request by naris-silpakit**
### What does this PR do?

- Updates the Prometheus helm chart to 15.3.0
- Updates the image tags to latest in readme example.
- Updates the selected version in the example in the Prometheus documentation.

### More

- [x] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/examples) to support my PR
- [x] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/ssp-eks-add-ons) repo (if applicable)
  - https://github.com/aws-samples/ssp-eks-add-ons/pull/12 
- [x] Yes, I have updated the [docs](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators
- [ ] E2E Test successfully complete before merge?

### Additional Notes
Testing notes:
  - created test directory and filled main.tf file with vpc, eks cluster with Prometheus and Amazon Managed Prometheus enabled
  - ran `terraform apply -auto-approve` and checked that deployment completes
  - connected to cluster and verified that nodes are online in `prometheus` namespace
  - checked prometheus pod logs to make sure there are no errors
  - connect to prometheus dashboard with `kubectl port-forward $(kubectl get pods --namespace=prometheus --selector='component=server' --output=name) 8080:9090 -n prometheus` to verify it loads

### CODE REVIEW

File: modules/kubernetes-addons/prometheus/locals.tf [hunk 1]
```diff
@@ -3,7 +3,7 @@ locals {
     name                       = "prometheus"
     chart                      = "prometheus"
     repository                 = "https://prometheus-community.github.io/helm-charts"
-    version                    = "14.6.0"
+    version                    = "15.3.0"
```
Comment (vara-bonthu) [inline modules/kubernetes-addons/prometheus/locals.tf:6]: @naris-silpakit Could you please find the all the occurrences of the Prometheus version and replace it? Some of the examples are also using the older version of Prometheus.

**Review by vara-bonthu — CHANGES_REQUESTED** [review]
LGTM! One change required

File: modules/kubernetes-addons/prometheus/locals.tf (hunk 1 above)
Comment (naris-silpakit) [reply modules/kubernetes-addons/prometheus/locals.tf:6]: Yep! I'll do that and push an update.

Comment (askulkarni2) [reply modules/kubernetes-addons/prometheus/locals.tf:6]: @naris-silpakit please mark this as resolved once you have addressed @vara-bonthu's comment.

**Review by askulkarni2 — COMMENTED** [review]
Looking good, waiting for comment to be addressed.

**Pull request merged**